src/components/action-bar/action-bar.spec.tsx

```diff
@@ -0,0 +1,184 @@
+                        <limel-action-bar-item isvisible=""></limel-action-bar-item>
+                    </div>
+                </mock:shadow-root>
+            </limel-action-bar>
+      `);
+    });
+
+    it('observes all actions in the intersection observer', () => {
+        expect(
+            intersectionObserverMockedFunctions.observe
+        ).toHaveBeenCalledTimes(5);
+    });
+
+    it('hides non-visable items in the menu', async () => {
+        triggerIntersection([true, true, false, false, false]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(2);
+        expect(hasOverflowMenu()).toBeTruthy();
+    });
+
+    it('shows the menu with some actions not intersecting', async () => {
+        triggerIntersection([true, true, false, false, false]);
+
+        await page.waitForChanges();
+        expect(hasOverflowMenu()).toBeTruthy();
+    });
+
+    it('does not show the menu when all actions intersect', async () => {
+        triggerIntersection([true, true, true, true, true]);
+
+        await page.waitForChanges();
+        expect(hasOverflowMenu()).toBeFalsy();
+    });
+
+    it('hides the menu when all actions intersect once more', async () => {
+        // Set some of the actions visible as a start
+        triggerIntersection([true, true, true, false, false]);
+
+        await page.waitForChanges();
+        expect(hasOverflowMenu()).toBeTruthy();
+
+        // Now add two more actions intersecting
+        triggerIntersection([true, true]);
+
+        // The menu should no longer be needed, all actions intersect
+        await page.waitForChanges();
+        expect(hasOverflowMenu()).toBeFalsy();
+    });
+
+    it('handles one more intersecting actions', async () => {
+        // Set some of the actions visible as a start
+        triggerIntersection([true, true, false, false, false]);
+
+        // Make one more item intersect
+        triggerIntersection([true]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(3);
+    });
+
+    it('handles one less intersecting actions', async () => {
+        // Set some of the actions visible as a start
+        triggerIntersection([true, true, true, true, false]);
+
+        // Make one less item intersect
+        triggerIntersection([false]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(3);
+    });
+
+    it('handles zero intersecting actions', async () => {
+        triggerIntersection([false, false, false, false, false]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(0);
+    });
+
+    it('handles all actions intersecting', async () => {
+        triggerIntersection([true, true, true, true, true]);
+
+        await page.waitForChanges();
+        expect(getNumVisibleActions()).toBe(5);
+    });
+
+    /**
+     * How can I get the disconnectedCallback lifecycle method to be called? //Pelle
```

Comment:
`disconnectedCallback` will only be invoked when the component is removed from the DOM. You can try something like this

```ts
page.body.removeChild(page.root)
```